Application crash: could not parse curse.json, panic

The application crashes, possibly while downloading the addon catalog.

Steps to reproduce the behavior:
1. Start the application.
2. Wait approximately 30 seconds.

**Software involved**
Please complete the following information:
 - OS: Windows 10 (1909)
 - Ajour version: 0.5.2
Note: Low bandwidth user - 1.5 Mbps download, may contribute to the issue.

**Log Output**
17:31:35.639 [ajour][INFO] Ajour 0.5.2 has started.
17:31:35.640 [ajour::gui][DEBUG] config loaded:
Config {
    wow: Wow {
        directory: Some(
            "C:\\Program Files (x86)\\World of Warcraft\\World of Warcraft",
        ),
        flavor: Retail,
    },
    addons: Addons {
        ignored: {
            Retail: [],
            RetailBeta: [],
        },
        release_channels: {
            RetailBeta: {},
            Retail: {
                "HandyNotes_SuramarTelemancy": Stable,
                "Ace3": Stable,
            },
        },
    },
    theme: None,
    column_config: V3 {
        my_addons_columns: [
            ColumnConfigV2 {
                key: "title",
                width: None,
                hidden: false,
            },
            ColumnConfigV2 {
                key: "local",
                width: Some(
                    150,
                ),
                hidden: false,
            },
            ColumnConfigV2 {
                key: "remote",
                width: Some(
                    150,
                ),
                hidden: false,
            },
            ColumnConfigV2 {
                key: "status",
                width: Some(
                    85,
                ),
                hidden: false,
            },
            ColumnConfigV2 {
                key: "channel",
                width: Some(
                    85,
                ),
                hidden: false,
            },
            ColumnConfigV2 {
                key: "author",
                width: Some(
                    85,
                ),
                hidden: true,
            },
            ColumnConfigV2 {
                key: "game_version",
                width: Some(
                    110,
                ),
                hidden: false,
            },
            ColumnConfigV2 {
                key: "date_released",
                width: Some(
                    110,
                ),
                hidden: true,
            },
            ColumnConfigV2 {
                key: "source",
                width: Some(
                    110,
                ),
                hidden: true,
            },
        ],
        catalog_columns: [
            ColumnConfigV2 {
                key: "addon",
                width: None,
                hidden: false,
            },
            ColumnConfigV2 {
                key: "description",
                width: Some(
                    150,
                ),
                hidden: false,
            },
            ColumnConfigV2 {
                key: "source",
                width: Some(
                    110,
                ),
                hidden: true,
            },
            ColumnConfigV2 {
                key: "num_downloads",
                width: Some(
                    105,
                ),
                hidden: true,
            },
            ColumnConfigV2 {
                key: "game_version",
                width: Some(
                    105,
                ),
                hidden: true,
            },
            ColumnConfigV2 {
                key: "date_released",
                width: Some(
                    105,
                ),
                hidden: false,
            },
            ColumnConfigV2 {
                key: "install",
                width: Some(
                    85,
                ),
                hidden: false,
            },
        ],
    },
    window_size: Some(
        (
            1004,
            677,
        ),
    ),
    scale: None,
    backup_directory: None,
    backup_addons: false,
    backup_wtf: false,
    hide_ignored_addons: false,
}
17:31:35.641 [ajour::gui][DEBUG] antialiasing: true
17:31:35.648 [ajour_core::theme][DEBUG] loading user themes
17:31:35.648 [ajour_core::utility][DEBUG] checking for application update
17:31:35.649 [ajour_core::fs::theme][DEBUG] loaded 0 user themes
17:31:36.285 [ajour::gui::update][DEBUG] Message::ThemesLoaded(0 themes)
17:31:36.285 [ajour::gui::update][DEBUG] Message::CachesLoaded(error: false)
17:31:36.285 [ajour::gui::update][DEBUG] Message::LatestRelease(Some("0.5.2"))
17:31:36.287 [ajour::gui::update][DEBUG] Message::Parse
17:31:36.287 [ajour::gui::update][DEBUG] preparing to parse addons in "C:\\Program Files (x86)\\World of Warcraft\\World of Warcraft\\_retail_\\Interface/AddOns"
17:31:36.288 [ajour::gui::update][DEBUG] preparing to parse addons in "C:\\Program Files (x86)\\World of Warcraft\\World of Warcraft\\_ptr_\\Interface/AddOns"
17:31:36.288 [ajour::gui::update][DEBUG] preparing to parse addons in "C:\\Program Files (x86)\\World of Warcraft\\World of Warcraft\\_beta_\\Interface/AddOns"
17:31:36.288 [ajour::gui::update][DEBUG] preparing to parse addons in "C:\\Program Files (x86)\\World of Warcraft\\World of Warcraft\\_classic_\\Interface/AddOns"
17:31:36.288 [ajour::gui::update][DEBUG] preparing to parse addons in "C:\\Program Files (x86)\\World of Warcraft\\World of Warcraft\\_classic_ptr_\\Interface/AddOns"
17:31:36.288 [ajour_core::parse][DEBUG] Retail - parsing addons folder
17:31:36.288 [ajour_core::parse][DEBUG] Retail PTR - parsing addons folder
17:31:36.288 [ajour_core::parse][DEBUG] Retail Beta - parsing addons folder
17:31:36.288 [ajour_core::parse][DEBUG] Classic - parsing addons folder
17:31:36.288 [ajour_core::parse][DEBUG] Classic PTR - parsing addons folder
17:31:36.289 [ajour_core::parse][DEBUG] Retail Beta - 0 folders in AddOns directory to parse
17:31:36.289 [ajour_core::parse][DEBUG] Retail - 134 folders in AddOns directory to parse
17:31:36.290 [ajour_core::parse][DEBUG] Retail - 134 fingerprints: 134 cached, 0 calculated, 0 added, 0 removed
17:31:36.290 [ajour::gui::update][ERROR] Message::ParsedAddons(Retail PTR) - Addon directory not found: "C:\\Program Files (x86)\\World of Warcraft\\World of Warcraft\\_ptr_\\Interface/AddOns"
17:31:36.290 [ajour::gui::update][ERROR] Message::ParsedAddons(Classic) - Addon directory not found: "C:\\Program Files (x86)\\World of Warcraft\\World of Warcraft\\_classic_\\Interface/AddOns"
17:31:36.290 [ajour::gui::update][ERROR] Message::ParsedAddons(Classic PTR) - Addon directory not found: "C:\\Program Files (x86)\\World of Warcraft\\World of Warcraft\\_classic_ptr_\\Interface/AddOns"
17:31:36.290 [ajour::gui::update][DEBUG] Message::ParsedAddons(Retail Beta, 0 addons)
17:31:36.300 [ajour_core::parse][DEBUG] Retail - 132 addon folders successfully parsed from '.toc'
17:31:36.300 [ajour_core::parse][DEBUG] Retail - 0 cached entries
17:31:36.300 [ajour_core::parse][DEBUG] Retail - 132 unique fingerprints to check against curse api
17:31:37.446 [ajour_core::catalog][DEBUG] Successfully fetched and parsed https://github.com/casperstorm/ajour-catalog/releases/latest/download/tukui.json
17:31:41.482 [ajour_core::parse][DEBUG] Retail - 87 curse packages fetched
17:31:41.482 [ajour_core::parse][DEBUG] Retail - 0 tukui packages fetched
17:31:42.140 [ajour_core::parse][DEBUG] Retail - 16 wowi packages fetched
17:31:42.140 [ajour_core::parse][DEBUG] Retail - 0 git packages fetched
17:31:42.141 [ajour_core::parse][DEBUG] Retail - 86 addons built from curse packages
17:31:42.141 [ajour_core::parse][DEBUG] Retail - 0 addons built from tukui packages
17:31:42.141 [ajour_core::parse][DEBUG] Retail - 0 addons built from wowi packages
17:31:42.141 [ajour_core::parse][DEBUG] Retail - 0 addons built from git packages
17:31:42.141 [ajour_core::parse][DEBUG] Retail - 2 unknown addon folders
17:31:42.141 [ajour_core::parse][DEBUG] Retail - 88 addons successfully parsed
17:31:42.143 [ajour::gui::update][DEBUG] Message::ParsedAddons(Retail, 88 addons)
17:31:59.707 [ajour_core::catalog][DEBUG] Successfully fetched and parsed https://github.com/casperstorm/ajour-catalog/releases/latest/download/wowi.json
17:32:05.653 [ajour_core::catalog][DEBUG] Could not parse https://github.com/casperstorm/ajour-catalog/releases/latest/download/curse.json
17:32:05.675 [ajour::gui::update][DEBUG] Message::CatalogDownloaded(6723 addons in catalog)
17:32:05.740 [panic][ERROR] thread 'main' panicked at 'called `Option::unwrap()` on a `None` value': src\gui\update.rs:1223
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: DefSubclassProc
  22: DefSubclassProc
  23: CallWindowProcW
  24: DispatchMessageW
  25: IsWindowVisible
  26: KiUserCallbackDispatcher
  27: NtUserDispatchMessage
  28: DispatchMessageW
  29: <unknown>
  30: <unknown>
  31: <unknown>
  32: <unknown>
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: <unknown>
  37: <unknown>
  38: <unknown>
  39: BaseThreadInitThunk
  40: RtlUserThreadStart

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,10 @@ and `Removed`.
 
 ## [Unreleased]
 
+### Fixed
+
+- Fix bug that would cause Ajour to panic if the catalog failed to download
+
 ## [0.5.2] - 2020-11-20
 
 ### Packaging

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,8 @@ and `Removed`.
 
 ### Fixed
 
+- Removed timeout for downloading the catalog. Users with slow internet can now
+  fetch the catalog regardless of how long it will take  
 - Fix bug that would cause Ajour to panic if the catalog failed to download
 
 ## [0.5.2] - 2020-11-20

--- a/crates/core/src/catalog.rs
+++ b/crates/core/src/catalog.rs
@@ -4,12 +4,9 @@ use crate::Result;
 use async_std::task;
 use chrono::prelude::*;
 use futures::future::join_all;
-use std::time::Duration;
 
 use isahc::{config::RedirectPolicy, prelude::*};
 use serde::Deserialize;
-
-const DEFAULT_TIMEOUT: u64 = 30;
 
 const CURSE_CATALOG_URL: &str =
     "https://github.com/casperstorm/ajour-catalog/releases/latest/download/curse.json";
@@ -24,7 +21,6 @@ pub async fn get_catalog_addons_from(url: &str) -> Result<Vec<CatalogAddon>> {
     let client = HttpClient::builder()
         .redirect_policy(RedirectPolicy::Follow)
         .max_connections_per_host(6)
-        .timeout(Duration::from_secs(DEFAULT_TIMEOUT))
         .build()
         .unwrap();
 

--- a/src/gui/update.rs
+++ b/src/gui/update.rs
@@ -1286,8 +1286,8 @@ pub fn handle_message(ajour: &mut Ajour, message: Message) -> Result<Command<Mes
             ajour.catalog_search_state.categories = ajour
                 .catalog_categories_per_source_cache
                 .get(&source.to_string())
-                .unwrap()
-                .to_vec();
+                .cloned()
+                .unwrap_or_default();
 
             ajour.catalog_search_state.category = CatalogCategory::All;
 

--- a/src/gui/update.rs
+++ b/src/gui/update.rs
@@ -1220,8 +1220,8 @@ pub fn handle_message(ajour: &mut Ajour, message: Message) -> Result<Command<Mes
             ajour.catalog_search_state.categories = ajour
                 .catalog_categories_per_source_cache
                 .get(&ajour.catalog_search_state.source.to_string())
-                .unwrap()
-                .to_vec();
+                .cloned()
+                .unwrap_or_default();
 
             ajour.catalog = Some(catalog);
 

